demisto_sdk/commands/pre_commit/hooks/validate_format.py

```diff
@@ -1,22 +1,24 @@
 from pathlib import Path
 from typing import Iterable, Optional
 
-from demisto_sdk.commands.pre_commit.hooks.hook import Hook, join_files
+from demisto_sdk.commands.pre_commit.hooks.hook import NIGHTLY_MODE, Hook, join_files
 
 
 class ValidateFormatHook(Hook):
     def prepare_hook(self, files_to_run: Optional[Iterable[Path]], **kwargs):
         """
         Prepares the Validate or the Format hook.
-        The default value is -g flag. In case of an input, we need to change it to -i, and add the input files.
+        In case of nightly mode and all files, runs validate/format with the --all flag, (nightly mode is not supported on specific files).
+        In case of an input or all files without nightly, runs validate/format on the given files.
+        Otherwise runs validate/format with the -g flag.
         Args:
             input_files (Optional[Iterable[Path]]): The input files to validate. Defaults to None.
         """
-        if self.input_mode:
+        if self.mode == NIGHTLY_MODE and self.all_files:
+            self.base_hook["args"].append("-a")
+        elif self.input_mode or self.all_files:
             self.base_hook["args"].append("-i")
```

Comment:
good idea. This way if running on regular mode we will run with `-i` on git files, which will identify more problems.